src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -521,31 +513,53 @@ public string Encoding
         /// </summary>
         [Category("Connection")]
         [Description("Determines the preferred PostgreSQL target server type.")]
-        [DisplayName("TargetSessionAttributes")]
+        [DisplayName("Target Session Attributes")]
         [NpgsqlConnectionStringProperty]
-        public string TargetSessionAttributes
+        public string? TargetSessionAttributes
         {
-            get => _targetSessionAttributes;
-            set
+            get => TargetSessionAttributesParsed switch
             {
-                if (value is null)
-                    value = PostgresEnvironment.TargetSessionAttributes ?? Npgsql.TargetSessionAttributes.Any.ToString();
-                if (!Enum.TryParse<TargetSessionAttributes>(value, out var result))
-                    throw new ArgumentException($"Unable to parse {nameof(TargetSessionAttributes)}", nameof(TargetSessionAttributes));
+                Npgsql.TargetSessionAttributes.Any           => "any",
+                Npgsql.TargetSessionAttributes.Primary       => "primary",
+                Npgsql.TargetSessionAttributes.Standby       => "standby",
+                Npgsql.TargetSessionAttributes.PreferPrimary => "prefer-primary",
+                Npgsql.TargetSessionAttributes.PreferStandby => "prefer-standby",
+                Npgsql.TargetSessionAttributes.ReadWrite     => "read-write",
+                Npgsql.TargetSessionAttributes.ReadOnly      => "read-only",
+                null => null,
 
-                _targetSessionAttributes = value;
-                SetValue(nameof(TargetSessionAttributes), value);
-                _targetSessionAttributesParsed = result;
+                _ => throw new ArgumentException($"Unhandled enum value '{TargetSessionAttributesParsed}'")
+            };
+
+            set
+            {
+                TargetSessionAttributesParsed = value is null ? null : ParseTargetSessionAttributes(value);
+                SetValue(nameof(TargetSessionAttributes), TargetSessionAttributesParsed);
```

Comment:
Shouldn't here be `SetValue(nameof(TargetSessionAttributes), value);` ?

Reply:
Yep, thanks.